### ISSUE TITLE
MixedPropertyTypeCoercion when instantiating SplFixedArray

### ISSUE THREAD

**Issue by villfa** [vimeo/psalm#7160]
Hi,

Here the problem I encounter: https://psalm.dev/r/d3add7ec8f

```php
class WithSplFixedArray
{
    /** @var SplFixedArray<int> */
    private $values;
    
    public function __construct(int $size)
    {
        $this->values = new SplFixedArray($size);
    }
}
```

```
INFO: MixedPropertyTypeCoercion - 20:25 - $this->values expects 'SplFixedArray<int>',  parent type `SplFixedArray<mixed>` provided
```

I think in this case SplFixedArray should be treated like a regular array and nothing should be raised.

**Comment by orklah** [comment]
TBH, I don't really know what are the rules on that and the reasons behind them. I saw some templates created as `empty` and other as `mixed`. In theory, `empty` one should behave better because empty is allowed value for any type (and covariance checks does not apply to empty). 
There must be a good reason behind why it's not done like that, but I don't know it :(

**Comment by AndrolGenhald** [comment]
I'm guessing they default to `mixed` to prevent false positives for stuff like [this](https://psalm.dev/r/ad84842374) ([this](https://psalm.dev/r/fb708d9cf0) gives `never` though), but I think defaulting to `never` might be a better choice in the long run.

Actually, I think maybe the best solution is to default to `never` but allow changing the type of local variables the same way it works for non-templates: https://psalm.dev/r/57800c0cb5. This should give the best of both worlds, types are as narrow as possible and there aren't false positives. I might try to look into this after my type system rewrite is finished.
